cluster-autoscaler/cloudprovider/azure/test/go.mod

```diff
@@ -1,87 +1,75 @@
 module k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/test
 
-go 1.23.0
+go 1.24.0
 
-toolchain go1.23.3
+toolchain go1.24.4
 
 require (
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.7.0
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5 v5.7.0
-	github.com/onsi/ginkgo/v2 v2.19.0
-	github.com/onsi/gomega v1.33.1
-	helm.sh/helm/v3 v3.15.2
-	k8s.io/api v0.30.2
-	k8s.io/apimachinery v0.30.2
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
-	sigs.k8s.io/controller-runtime v0.18.4
+	github.com/onsi/ginkgo/v2 v2.23.4
+	github.com/onsi/gomega v1.37.0
+	helm.sh/helm/v3 v3.18.3
+	k8s.io/api v0.34.0-alpha.0
```

Comment:
Happy to punt on this for now, but going forward should we try to pin k8s.io modules to GA versions?